File: GaReGe.server/GaReGe.server/Entity/Vehicle.cs
namespace GaReGe.server.Entity {
    public class Vehicle {
        public int VehicleId { get; set; }
        public string LicensePlate { get; set; } = string.Empty;
        public string Color { get; set; } = string.Empty;
        public string Brand { get; set; } = string.Empty;
        public string Model { get; set; } = string.Empty;
        public int NumWheels { get; set; }

        // FK
        public int VehicleTypeId { get; set; }
        public int MemberId { get; set; }

        // nav props
        public VehicleType MyProperty { get; set; }
        public Member Member { get; set; }

    }
}

File: GaReGe.client/src/types/vehicle.ts
export type VehicleSummary = {
    vehicleId: number,
    licensePlate: string;
    brand: string;
    model: string;
    vehicleTypeName: string;
    memberName: string;
}


export type VehicleDetail = {
    vehicleId: number; 
    licensePlate: string;
    color: string;
    brand: string;
    model: string;
    numWheels: number;
    memberId: number;
    vehicleTypeId: number;
    vehicleTypeName: string;
    memberName: string;
}


// export type Vehicle = {
//     vehicleId?: number;
//     licensePlate: string;
//     color: string;
//     brand: string;
//     model: string;
//     numWheels: number;
//     memberId: number;
//     vehicleTypeId: number;
//     vehicleTypeName: string;
//     memberName: string;
// }